libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -247,31 +253,31 @@ class MapboxTripSession(
     }
 
     private fun updateRawLocation(rawLocation: Location) {
+        locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
+        lastLocationUpdateTimeMillis = Date().time
+        statusUpdateTimer.stopJobs()
         ioJobController.scope.launch {
-            rawLocation.let {
-                navigator.updateLocation(it)
-            }
+            navigator.updateLocation(rawLocation)
+            updateDataFromNavigatorStatus()
         }
-        locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
-        if (this.rawLocation == null) {
-            fireOffStatusPolling()
+        statusUpdateTimer.startTimer {
+            if (Date().time - lastLocationUpdateTimeMillis >= STATUS_POLLING_INTERVAL) {
```

Comment:
This extra check shouldn't be needed.

Reply:
But what in case if we will change this values? I'm agree that it is will always true for case when we have 1s for this check and 1s for timer... but if we change one of this values?

Reply:
oh i just noticed it was already asked here https://github.com/mapbox/mapbox-navigation-android/pull/2719#discussion_r404911308

this is a good thing to show with unit tests, because many devs will look at in and not see the reasoning